index.d.ts

```diff
@@ -1946,87 +1986,304 @@ declare namespace Eris {
+    /**
+     * Disconnects the shard
+     * @arg options Shard disconnect options
+     */
+    disconnect(options?: {
+      /** False means destry everything, true means you want to reconnect in the future, "auto" will autoreconnect */
+      reconnect: "auto" | boolean;
+    }): void;
+    /**
+     * Update the bot's AFK status. Setting this to true will enable push notifications for userbots.
+     * @param afk Whether the bot user is AFK or not
+     */
     editAFK(afk: boolean): void;
-    editStatus(status?: Status, game?: ActivityPartial<BotActivityType>): void;
+    /**
+     * Updates the bot's status on all guilds the shard is in
+     * @arg status Sets the bot's status, or set's the bot's active game
+     * @arg game Sets the bot's active game, null to clear
+     */
+    editStatus(status?: Status | ActivityPartial<BotActivityType>, game?: ActivityPartial<BotActivityType>): void;
     on: ShardEvents<this>;
+    resume(): void;
     toString(): string;
     toJSON(props?: string[]): JSONCache;
     sendWS(op: number, _data: object, priority: boolean): void;
   }
 
-  export class Command {
-    subcommands: { [s: string]: Command };
-    subcommandAliases: { [alias: string]: Command };
-    label: string;
-    parentCommand?: Command;
-    description: string;
-    fullDescription: string;
-    usage: string;
+  interface LatencyRef {
+    /** Interval between consuming tokens */
+    latency: number;
+    offset: number;
+    raw: number[];
+    timeOffset: number;
+    timeOffsets: number[];
+    lastTimeOffsetCheck: number;
+  }
+
+  /**
+   * Ratelimit requests and release in sequence
+   */
+  class SequentialBucket {
+    /** How many tokens the bucket can consume in the current interval */
+    limit: number;
+    /** Whether the queue is being processed */
+    processing: boolean;
+    /** How many tokens the bucket has left in the current interval */
+    remaining: number;
+    /** Timestamp of next reset */
+    reset: number;
+    latencyRef: LatencyRef;
+    /**
+     * Construct a SequentialBucket
+     * @arg limit THe max number of tokens the bucket can consume per interval
+     * @arg latencyRef An object
+     */
+    constructor(limit: number, latencyRef?: LatencyRef);
+    /**
+     * Queue something in the SequentialBucket
+     * @param func A function to call when a token can be consumed. The function will be passed a callback argument, which must be called to allow the bucket to continue to work
+     */
+    queue(func: Function, short?: boolean): void;
+    check(override?: boolean): void;
+  }
+
+  type RequestMethod = "GET" | "PATCH" | "DELETE" | "POST" | "PUT";
+  interface File {
+    /** A buffer containing file data */
+    file: Buffer;
+    /** What to name the file */
+    name: string;
+  }
+
+  /**
+   * Handles API requests
+   */
+  class RequestHandler implements SimpleJSON {
+    agent: unknown; // HTTP Agent - Do we parse our preferred agent here?
```

Comment:
cc @abalabahaha need clarification on this please
This is referenced from `Client.options.agent`